### ISSUE TITLE
Python 3 clipboard encoding

### ISSUE THREAD

**Issue by tbnorth** [leo-editor/leo-editor#147]
Leo 5.0-final, build 20150304120302, Wed Mar  4 12:03:02 CST 2015
Git repo info: branch = master, commit = 0b4696be5c18
Python 2.7.6, PyQt version 4.8.6
linux2

HOME=/tmp python3 launchLeo.py

Text: ÷ Σ

In body text copy / paste (Ctrl-C, Ctrl-V) works as expected in py2, but there's a Σ -> £ translation on paste in py3

**Comment by edreamleo** [comment]
Thanks for this report.  This must be fixed for 5.1 b1.

**Comment by edreamleo** [comment]
Rev d89d4bc attempted a fix, but the fix didn't work.  This bug happens only on Linux...

**Comment by edreamleo** [comment]
This looks like a PyQt or Qt bug in Python 3, and only on Linux.

The call to QTextBrowser.toPlainText in qtew.getAllText returns a *unicode* (Python 3 str) string with a value different from what is actually shown in the body pane!  Because the string is already unicode, Leo can do nothing more.

A truly horrible workaround is possible for the body pane only, namely using p.b instead of w.get (calls getAllText) in leoFrame.copyText/cutText, but I would rather not do that.  It's the worst kind of hack, leaves the bug in place for all other widgets, and must be revisited later when the real bug is fixed.

I'll investigate further.   I shall not delay 5.1b1 for this.  It's presumably been around a long time and surely has nothing to do with recent changes.

**Comment by edreamleo** [comment]
I've just reported the bug to the Qt people: https://bugreports.qt.io/browse/QTBUG-44822
And also to the pyqt mailing list: pyqt@riverbankcomputing.com, archives at http://www.riverbankcomputing.com/pipermail/pyqt/

**Comment by edreamleo** [comment]
Got an instant comment from the pyqt people::

Have you tried with more recent PyQt/Qt versions?

I can't reproduce this on Qt 5.3.2 on Debian Jessie with python3:

    from PyQt5.QtCore import QT_VERSION_STR
    from PyQt5.QtWidgets import QApplication, QTextBrowser
    a = QApplication([])
    tb = QTextBrowser()
    tb.setText('÷ Σ')
    print(QT_VERSION_STR)
    print(tb.toPlainText())

-> prints 5.3.2 and ÷ Σ

**Comment by edreamleo** [comment]
OMG, this is a big big problem.  Reselecting a node will change Σ into £ !

**Comment by largo84** [comment]
Hmm, wonder why this is just a Linux problem. I don't see that on Windows.

**Comment by edreamleo** [comment]
@largo84 Right.  It's only a Linux problem, and only for "older" versions of pyqt such as PyQt 5.2.1.  Having said that, many people will be affected.

In the meantime, I really can't recommend using Leo with Python 3 on Ubuntu.

**Comment by edreamleo** [comment]
I am going to remove any target from this bug.  We may have to wait 6 months for this to be fixed in widely used linux distributions.

**Comment by edreamleo** [comment]
[This thread](https://groups.google.com/d/msg/leo-editor/lvtpbYRP_qE/4wTwyneDBwAJ) may be relevant.  Or not.

**Comment by edreamleo** [comment]
I am going to change the label from wontfix to can'tfix, indicating that we are waiting on a bug fix outside of Leo.
